src/Servant/Auth/Hmac/Crypto.hs

```diff
@@ -131,16 +132,19 @@ that given 'Request' has header:
 Authentication: HMAC <signature>
 @
 
-It checks whether @<signature>@ is true request signature.
+It checks whether @<signature>@ is true request signature. Function returns 'Nothing'
+if it is true, and 'Just' error message otherwise.
 -}
 verifySignatureHmac
     :: (SecretKey -> ByteString -> Signature)  -- ^ Signing function
     -> SecretKey  -- ^ Secret key that was used for signing 'Request'
     -> RequestPayload
-    -> Bool
+    -> Maybe LBS.ByteString
 verifySignatureHmac signer sk signedPayload = case unsignedPayload of
-    Nothing         -> False
-    Just (pay, sig) -> sig == requestSignature signer sk pay
+    Nothing         -> Just "Can not extract auth header"
+    Just (pay, sig) -> if sig == requestSignature signer sk pay
+        then Nothing
+        else Just "Signatures don't match"
   where
     -- Extracts HMAC signature from request and returns request with @authHeaderName@ header
     unsignedPayload :: Maybe (RequestPayload, Signature)
```

Comment:
Could you please change type of `unsignedPayload` from `Maybe` to `Either LBS.ByteString Maybe (RequestPayload, Signature)`. Just to give better error message: there can't be no header with named `authHeaderName` or header value might not start with `HMAC ` prefix.